Kernel Panic on RTMP Stream Opening

## Which version are you using?

master

## Which operating system are you using?

<!-- fill checkboxes with a x. Example: [x] Linux -->

- [x] macOS arm64 standard

## Describe the issue

```
2023/02/08 15:07:51 INF [RTSP] listener opened on :8554 (TCP), :8000 (UDP/RTP), :8001 (UDP/RTCP)
2023/02/08 15:07:51 INF [RTMP] listener opened on :1935
2023/02/08 15:07:51 INF [HLS] listener opened on :8888
2023/02/08 15:07:51 INF [WebRTC] listener opened on :8889 (HTTP)
2023/02/08 15:07:59 INF [RTMP] [conn 192.168.1.28:55154] opened
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x104e22240]

goroutine 24 [running]:
github.com/aler9/gortsplib/v2/pkg/format.(*H264).PayloadType(0x4645bee4e3eda3c7?)
        /Users/xaviergeerinck/Projects/rtsp-simple-server/rtsp-gortsplib/pkg/format/h264.go:95
github.com/aler9/gortsplib/v2.newServerStreamMedia(0x140004f08a0, 0x1400004a190)
        /Users/xaviergeerinck/Projects/rtsp-simple-server/rtsp-gortsplib/server_stream_media.go:41 +0x2b8
github.com/aler9/gortsplib/v2.NewServerStream({0x140003780e0, 0x2, 0x2})
        /Users/xaviergeerinck/Projects/rtsp-simple-server/rtsp-gortsplib/server_stream.go:42 +0x130
github.com/aler9/rtsp-simple-server/internal/core.newStream({0x140003780e0?, 0x10495a59c?, 0x14000451718?}, 0x8?, 0x140002fe288)
        /Users/xaviergeerinck/Projects/rtsp-simple-server/internal/core/stream.go:24 +0x28
github.com/aler9/rtsp-simple-server/internal/core.(*path).sourceSetReady(0x14000502820, {0x140003780e0?, 0x140004015c0?, 0x140004518ba?}, 0xf?)
        /Users/xaviergeerinck/Projects/rtsp-simple-server/internal/core/path.go:640 +0x3c
github.com/aler9/rtsp-simple-server/internal/core.(*path).handlePublisherStart(0x14000502820, {{0x1053b93b8, 0x1400001a000}, {0x140003780e0, 0x2, 0x2}, 0x1, 0x1400058e300})
        /Users/xaviergeerinck/Projects/rtsp-simple-server/internal/core/path.go:788 +0xd8
github.com/aler9/rtsp-simple-server/internal/core.(*path).run.func2(0x14000502820)
        /Users/xaviergeerinck/Projects/rtsp-simple-server/internal/core/path.go:473 +0x6b8
github.com/aler9/rtsp-simple-server/internal/core.(*path).run(0x14000502820)
        /Users/xaviergeerinck/Projects/rtsp-simple-server/internal/core/path.go:499 +0x27c
created by github.com/aler9/rtsp-simple-server/internal/core.newPath
        /Users/xaviergeerinck/Projects/rtsp-simple-server/internal/core/path.go:283 +0x570
exit status 2
```

## Describe how to replicate the issue

Start the server with the below:

```
go run .
```

Opened an RTMP stream through the DJI Streaming

## Did you attach the server logs?

yes

## Did you attach a network dump?

<!--
If the bug arises when using rtsp-simple-server with an external hardware or software, the most helpful content you can provide is a dump of the data exchanged between the server and the target (network dump), that can be generated in this way:
1) Download wireshark (https://www.wireshark.org/)
2) Start capturing on the interface used for exchanging RTSP (if the server and the target software are both installed on your pc, the interface is probably "loopback", otherwise it's the one of your network card)
3) Start the server and replicate the issue
4) Stop capturing, save the result in .pcap format
5) Attach
-->

yes

[capture_rtmp_rtsp.pcapng.zip](https://github.com/aler9/rtsp-simple-server/files/10686519/capture_rtmp_rtsp.pcapng.zip)

--- a/internal/rtmp/conn.go
+++ b/internal/rtmp/conn.go
@@ -584,7 +584,7 @@ func (c *Conn) WriteMessage(msg message.Message) error {
 	return c.mrw.Write(msg)
 }
 
-func trackFromH264DecoderConfig(data []byte) (*format.H264, error) {
+func trackFromH264DecoderConfig(data []byte) (format.Format, error) {
 	var conf h264conf.Conf
 	err := conf.Unmarshal(data)
 	if err != nil {
@@ -766,9 +766,9 @@ func (c *Conn) readTracksFromMetadata(payload []interface{}) (format.Format, *fo
 	}
 }
 
-func (c *Conn) readTracksFromMessages(msg message.Message) (*format.H264, *format.MPEG4Audio, error) {
+func (c *Conn) readTracksFromMessages(msg message.Message) (format.Format, *format.MPEG4Audio, error) {
 	var startTime *time.Duration
-	var videoTrack *format.H264
+	var videoTrack format.Format
 	var audioTrack *format.MPEG4Audio
 
 	// analyze 1 second of packets

--- a/internal/rtmp/conn_test.go
+++ b/internal/rtmp/conn_test.go
@@ -5,6 +5,7 @@ import (
 	"net"
 	"net/url"
 	"testing"
+	"time"
 
 	"github.com/aler9/gortsplib/v2/pkg/codecs/h264"
 	"github.com/aler9/gortsplib/v2/pkg/codecs/mpeg4audio"
@@ -537,13 +538,28 @@ func TestReadTracks(t *testing.T) {
 			},
 		},
 		{
-			"missing metadata",
+			"missing metadata, video+audio",
 			&format.H264{
 				PayloadTyp:        96,
 				SPS:               sps,
 				PPS:               pps,
 				PacketizationMode: 1,
 			},
+			&format.MPEG4Audio{
+				PayloadTyp: 96,
+				Config: &mpeg4audio.Config{
+					Type:         2,
+					SampleRate:   44100,
+					ChannelCount: 2,
+				},
+				SizeLength:       13,
+				IndexLength:      3,
+				IndexDeltaLength: 3,
+			},
+		},
+		{
+			"missing metadata, audio",
+			nil,
 			&format.MPEG4Audio{
 				PayloadTyp: 96,
 				Config: &mpeg4audio.Config{
@@ -796,6 +812,7 @@ func TestReadTracks(t *testing.T) {
 					SPS: sps,
 					PPS: pps,
 				}.Marshal()
+
 				err = mrw.Write(&message.MsgVideo{
 					ChunkStreamID:   message.MsgVideoChunkStreamID,
 					MessageStreamID: 0x1000000,
@@ -811,6 +828,7 @@ func TestReadTracks(t *testing.T) {
 					ChannelCount: 2,
 				}.Marshal()
 				require.NoError(t, err)
+
 				err = mrw.Write(&message.MsgAudio{
 					ChunkStreamID:   message.MsgAudioChunkStreamID,
 					MessageStreamID: 0x1000000,
@@ -855,6 +873,7 @@ func TestReadTracks(t *testing.T) {
 					SPS: sps,
 					PPS: pps,
 				}.Marshal()
+
 				err = mrw.Write(&message.MsgVideo{
 					ChunkStreamID:   message.MsgVideoChunkStreamID,
 					MessageStreamID: 0x1000000,
@@ -893,6 +912,7 @@ func TestReadTracks(t *testing.T) {
 					SPS: sps,
 					PPS: pps,
 				}.Marshal()
+
 				err = mrw.Write(&message.MsgVideo{
 					ChunkStreamID:   message.MsgVideoChunkStreamID,
 					MessageStreamID: 0x1000000,
@@ -908,6 +928,7 @@ func TestReadTracks(t *testing.T) {
 					ChannelCount: 2,
 				}.Marshal()
 				require.NoError(t, err)
+
 				err = mrw.Write(&message.MsgAudio{
 					ChunkStreamID:   message.MsgAudioChunkStreamID,
 					MessageStreamID: 0x1000000,
@@ -919,11 +940,12 @@ func TestReadTracks(t *testing.T) {
 				})
 				require.NoError(t, err)
 
-			case "missing metadata":
+			case "missing metadata, video+audio":
 				buf, _ := h264conf.Conf{
 					SPS: sps,
 					PPS: pps,
 				}.Marshal()
+
 				err = mrw.Write(&message.MsgVideo{
 					ChunkStreamID:   message.MsgVideoChunkStreamID,
 					MessageStreamID: 0x1000000,
@@ -939,6 +961,7 @@ func TestReadTracks(t *testing.T) {
 					ChannelCount: 2,
 				}.Marshal()
 				require.NoError(t, err)
+
 				err = mrw.Write(&message.MsgAudio{
 					ChunkStreamID:   message.MsgAudioChunkStreamID,
 					MessageStreamID: 0x1000000,
@@ -947,6 +970,37 @@ func TestReadTracks(t *testing.T) {
 					Channels:        flvio.SOUND_STEREO,
 					AACType:         flvio.AAC_SEQHDR,
 					Payload:         enc,
+				})
+				require.NoError(t, err)
+
+			case "missing metadata, audio":
+				enc, err := mpeg4audio.Config{
+					Type:         2,
+					SampleRate:   44100,
+					ChannelCount: 2,
+				}.Marshal()
+				require.NoError(t, err)
+
+				err = mrw.Write(&message.MsgAudio{
+					ChunkStreamID:   message.MsgAudioChunkStreamID,
+					MessageStreamID: 0x1000000,
+					Rate:            flvio.SOUND_44Khz,
+					Depth:           flvio.SOUND_16BIT,
+					Channels:        flvio.SOUND_STEREO,
+					AACType:         flvio.AAC_SEQHDR,
+					Payload:         enc,
+				})
+				require.NoError(t, err)
+
+				err = mrw.Write(&message.MsgAudio{
+					ChunkStreamID:   message.MsgAudioChunkStreamID,
+					MessageStreamID: 0x1000000,
+					Rate:            flvio.SOUND_44Khz,
+					Depth:           flvio.SOUND_16BIT,
+					Channels:        flvio.SOUND_STEREO,
+					AACType:         flvio.AAC_SEQHDR,
+					Payload:         enc,
+					DTS:             1 * time.Second,
 				})
 				require.NoError(t, err)
 
@@ -1017,6 +1071,7 @@ func TestReadTracks(t *testing.T) {
 					ChannelCount: 2,
 				}.Marshal()
 				require.NoError(t, err)
+
 				err = mrw.Write(&message.MsgAudio{
 					ChunkStreamID:   message.MsgAudioChunkStreamID,
 					MessageStreamID: 0x1000000,

